recipes-bsp/u-boot/u-boot-distro-boot.bbappend

```diff
@@ -10,12 +10,80 @@ SRC_URI:append = " \
 # representing device-tree overlays) to be used when booting.
 FITCONF_FDT_OVERLAYS ??= ""
 
+# Machine specific values related to HAB/AHAB fusing (i.MX specific)
+FUSE_BANK = ""
+FUSE_WORD = ""
+FUSE_NUM = ""
+STATUS_CMD = ""
+CLOSE_CMD = ""
+
+FUSE_BANK:mx6ull-generic-bsp = "3"
+FUSE_WORD:mx6ull-generic-bsp = "0"
+FUSE_NUM:mx6ull-generic-bsp = "1 2 3 4 5 6 7 8"
+STATUS_CMD:mx6ull-generic-bsp = "hab_status"
+CLOSE_CMD:mx6ull-generic-bsp = "fuse prog -y 0 6 0x00000002"
+
+FUSE_BANK:mx6q-generic-bsp = "3"
+FUSE_WORD:mx6q-generic-bsp = "0"
+FUSE_NUM:mx6q-generic-bsp = "1 2 3 4 5 6 7 8"
+STATUS_CMD:mx6q-generic-bsp = "hab_status"
+CLOSE_CMD:mx6q-generic-bsp = "fuse prog -y 0 6 0x00000002"
+
+FUSE_BANK:mx6dl-generic-bsp = "3"
+FUSE_WORD:mx6dl-generic-bsp = "0"
+FUSE_NUM:mx6dl-generic-bsp = "1 2 3 4 5 6 7 8"
+STATUS_CMD:mx6dl-generic-bsp = "hab_status"
+CLOSE_CMD:mx6dl-generic-bsp = "fuse prog -y 0 6 0x00000002"
+
+FUSE_BANK:mx7-generic-bsp = "6"
+FUSE_WORD:mx7-generic-bsp = "0"
+FUSE_NUM:mx7-generic-bsp = "1 2 3 4 5 6 7 8"
+STATUS_CMD:mx7-generic-bsp = "hab_status"
+CLOSE_CMD:mx7-generic-bsp = "fuse prog -y 1 3 0x02000000"
+
+FUSE_BANK:mx8m-generic-bsp = "6"
+FUSE_WORD:mx8m-generic-bsp = "0"
+FUSE_NUM:mx8m-generic-bsp = "1 2 3 4 5 6 7 8"
+STATUS_CMD:mx8m-generic-bsp = "hab_status"
+CLOSE_CMD:mx8m-generic-bsp = "fuse prog -y 1 3 0x02000000"
+
+FUSE_BANK:mx8qm-generic-bsp = "0"
+FUSE_WORD:mx8qm-generic-bsp = "722"
+FUSE_NUM:mx8qm-generic-bsp = "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16"
+STATUS_CMD:mx8qm-generic-bsp = "ahab_status"
+CLOSE_CMD:mx8qm-generic-bsp = "ahab_close"
+
+FUSE_BANK:mx8x-generic-bsp = "0"
+FUSE_WORD:mx8x-generic-bsp = "730"
+FUSE_NUM:mx8x-generic-bsp = "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16"
+STATUS_CMD:mx8x-generic-bsp = "ahab_status"
+CLOSE_CMD:mx8x-generic-bsp = "ahab_close"
+
+VARS_TO_DEL="$(cat ${WORKDIR}/uEnv.txt.in | grep -o -P '^\S*[a-z]\S{0,}=' | cut -d"=" -f1 | tr '\n' ' ') \
+    kernel_image \
+    ramdisk_image \
+    fdtdir \
+    bootargs \
+    kernel_image2 \
+    ramdisk_image2 \
+    fdtdir2 \
+    bootargs2 \
+    fdtfile2 \
+    temp \
+"
```

Comment:
So, actually I originally did have `VARS_TO_DEL` in `do_compile()`. But then the behavior was weird for some reason.

I had something like:

```
do_compile:append() {
    VARS_TO_DEL="test"

[...]

    sed ... \
        -e "s/@@VARS_TO_DEL@@/${VARS_TO_DEL}/" \
}
```

But the resulting `uEnv.txt` file always had `vars_to_del=`, as if the variable was empty. I even added a `bbnote` before the `sed` to print out `VARS_TO_DEL` and it would show as ""test" in the logs. But when it came to the `sed` it was as if the variable was not set to anything. I only ever got the `sed` to work when I defined the variable outside of  `do_compile()`. Eventually I stopped trying to get it to work in `do_compile()` since I was spending too much time on debugging this weird behavior. 

I can give it a try again to implement this suggestion, but it was really strange how variables set in `do_compile()` were behaving here.

Reply:
Interesting...

When you had the code you mentioned above, you didn't have the variable also set in bitbake, right? If you had, it would explain the behavior since the bitbake variable replacement would have precedence over the shell variables (something that could be avoided by using writing `-e "s/@@VARS_TO_DEL@@/$VARS_TO_DEL/"` in the `sed` command).

I believe the simplest way to debug it is by checking the code generated by bitbake e.g. by running `bitbake virtual/bootloader -e` and checking the contents of `do_compile()`.

Anyways, let's try to solve this. Just let me know if you need help debugging it.

Reply:
Huh your regex works, until it gets substituted into the main `sed` command:

```
sed: -e expression #9, char 35: unterminated `s' command
```

I guess the `sed` command in the command substitution is interfering with the main/outer `sed` command. I'm not sure how to modify the inner `sed` command in a way that is compatible with the outer command while still retaining the same regex functionality. The closest I got was to remove the ending "gp" options. This doesn't give an error, but obviously breaks the intended result.

Any idea how to adjust the expression here to work in the main `sed` command?

Either that or we just manually list out the variables in `uEnv.txt.in` instead of trying to programatically generate an initial list. Anyways we already have to manually list out the other variables that aren't initially in `uEnv.txt.in`.

Reply:
Running the command manually I found that `vars_to_del` has newlines in it which is confusing `sed`. Here's a possible fix:

```
    vars_to_del=$(cat ${WORKDIR}/uEnv.txt.in |
                  sed -n -e 's/^\([A-Za-z_][A-Za-z0-9_]*\)=.*/\1/gp')
    # Get rid of newlines:
    vars_to_del=$(echo ${vars_to_del})
```

Reply:
One thing to point that though: there is no inner/outer sed; the value of `vars_to_del` is simply set before the second invocation of the `sed` command.

Reply:
Done in V2